lib/active_fulfillment/services/amazon_mws.rb

```diff
@@ -217,26 +193,12 @@ def message_from(response)
 
     ## PARSING
 
-    def parse_response(service, op, xml)
+    def parse_document(xml)
       begin
         document = REXML::Document.new(xml)
       rescue REXML::ParseException
         return { :success => FAILURE }
       end
-
-      case service
```

Comment:
This was particularly hard to decipher. Matching up response parsing with the request via keying into a hash defined elsewhere seems cumbersome.